Assembler.UnitTests/FrameHandlers/InitialFrameHandlerTests.cs

```diff
@@ -88,6 +89,9 @@ public void Handle_MessageInCacheAndMiddleReceived_ReleasesTheOldOneWithoutEnric
             _dateTimeProviderMock.Verify(provider => provider.Now, Times.Once);
             Assert.AreEqual(DateTime.MinValue, newMessage.LastFrameReceived);
 
+            Assert.AreEqual(frame.Object.Guid, newMessage.BasedOnGuids.Single());
```

Comment:
Its a bad practice to use auto-generated guid in UT because every time a new guid will be generated. I prefer this arrange:

![image](https://user-images.githubusercontent.com/62211836/78647365-2c92de80-78c3-11ea-8119-9689054b3a3e.png)


And extract the generation of BaseFrame to TestUtiltities. Do the same in this class, and Middle and Final tests classes

Reply:
I don't get why this is bad practice, why do you care if a new guid is created every time.

Reply:
Because UT should be static, it shouldnt use any different data in every run. Although new guid shouldnt affect this UT, I think we should avoid it whenever we can

Reply:
f